Sources/DifferentiableSwiftExamplesDocumentation/DifferentiableSwiftExamples.docc/Tutorials/GradientDescent.tutorial

```diff
@@ -1,9 +1,75 @@
 @Tutorial(time: 20) {
-    @Intro(title: "Gradient Descent") {
-        # Coming Soon
+    @Intro(title: "Gradient descent") {
+        Gradient descent optimization is a powerful technique for optimizing arbitrary parameters in a complex system to arrive at a desired result.
+        
+        This tutorial will demonstrate the use of gradient descent to train a perceptron to perform a boolean logic function.
     }
     
-    @Section(title: "Coming Soon") {
-        
+    @Section(title: "Training a perceptron") {
+        @ContentAndMedia {
+            Train a basic perceptron to mimic an AND gate.
+            
+            A perceptron is a simple, single-level neural network. In this case, we'll build one with two neurons, one for each input in the AND gate. The neurons will have trainable weights and a bias value, along with a nonlinear activation function.
+            
+            This perceptron will be trained against the desired behavior of an AND gate, and weights for the neurons will be optimized via gradient descent.
+        }
+        @Steps {
+            @Step {
+                Differentiable Swift is an experimental language feature. To begin with, we must enable it by importing the `_Differentiation` module.
+                @Code(name: "main.swift", file: "GradientDescent-01-01.swift")
+            }
+            @Step {
+                Let's create the perceptron as a custom data type. The weights for the connections between neurons will be randomly-initialized floating point values, and the bias will start at zero.
+                @Code(name: "main.swift", file: "GradientDescent-01-02.swift")
+            }
+            @Step {
+                Because all of the properties of our perceptron conform to `Differentiable`, it's easy to make the perceptron itself differentiable by adding the conformance to `Differentiable`.
+                @Code(name: "main.swift", file: "GradientDescent-01-03.swift")
+            }
+            @Step {
+                A convenient language feature that was introduced to Swift is the ability of structs to be called as if they were functions. To do so, you need to define a `callAsFunction()` method on the struct.
```

Comment:
I feel adding the callAsFunction here is adding unnecessary complexity. I'm guessing this is inspired from a more pythony way of training a model/inferencing? I'd rather have a named function to avoid the added callAsFunction complexity.

Reply:
Feels very Pythony to me which I understand but I don't necessarily see the value we add by doing this and/or promoting this as a good approach. But I can accept it!